### PR TITLE
Always print masterserver response request id

### LINKED ISSUE #606: Always log request id with master server
Always log the request id in all error messages related to interactions with master server.

Makes it way easier to debug stuff when you don't have to go hunting for the matching entry in master server logs.

### PR DESCRIPTION
**Pull request by GeckoEidechse**
Always print masterserver response request id to logs.

Worked when I tested it myself:
image

Closes #606

### CODE REVIEW

File: primedev/masterserver/masterserver.cpp
Comment (ASpoonPlaysGames) [inline primedev/masterserver/masterserver.cpp:1]: This wont cause conflicts with my logging PR but will break it :)

So uhh, ping me when/if this gets merged

**Review by ASpoonPlaysGames — APPROVED** [review]
Code looks good

File: primedev/masterserver/masterserver.cpp [hunk 1]
```diff
@@ -1285,6 +1285,16 @@ void MasterServerPresenceReporter::InternalAddServer(const ServerPresence* pServ
...
 				}
 
+				// Log request id for easier debugging when combining with logs on masterserver
+				if (serverAddedJson.HasMember("id"))
+				{
+					spdlog::info("Request id: {}", serverAddedJson["id"].GetString());
+				}
+				else
+				{
+					spdlog::error("No request id? WTF (put a sensible error message here)");
```
Comment (GeckoEidechse) [inline primedev/masterserver/masterserver.cpp:1295]:
```suggestion
					spdlog::error("Couldn't find request id in response");
```

File: primedev/masterserver/masterserver.cpp
Comment (GeckoEidechse) [reply primedev/masterserver/masterserver.cpp:1]: Changed in 14e7ec46bb6dc29ddc6c77356b341fe4f710dba9

File: primedev/masterserver/masterserver.cpp [hunk 2]
```diff
@@ -1281,6 +1281,16 @@ void MasterServerPresenceReporter::InternalAddServer(const ServerPresence* pServ
...
```
[the remark this answers is not in the document]
Comment (GeckoEidechse) [reply primedev/masterserver/masterserver.cpp]: Changed in 14e7ec46bb6dc29ddc6c77356b341fe4f710dba9

**Comment by GeckoEidechse** [comment]
Tested myself I guess and will test the rest in prod :P

**Pull request merged**